### ISSUE TITLE
cvxopt won't install on Windows

### ISSUE THREAD

**Issue by stormont** [cvxopt/cvxopt#114]
Using Win 10 and Python 3.6.5. Install `cvxopt` fails with this message:

```
  ...

  UPDATING build\lib.win-amd64-3.6\cvxopt/_version.py
  set build\lib.win-amd64-3.6\cvxopt/_version.py to '1.1.9'
  running build_ext
  building 'base' extension
  error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

  ----------------------------------------
  Failed building wheel for cvxopt
  Running setup.py clean for cvxopt

  ...

  building 'base' extension
  error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools
```

Except, I do have Microsoft Visual C++ 14.0 installed. I've also tried uninstalling it, reinstalling it, restarted my system, installed both the x64 and x86 variants... `PATH` variables all look fine. `cvxopt` doesn't seem to correctly identify the install and fails.

**Comment by martinandersen** [comment]
The problem is probably that Microsoft's C compiler does not recognize the standard `double complex`. See #94 and [C complex math support](https://docs.microsoft.com/en-us/cpp/c-runtime-library/complex-math-support).

**Comment by stormont** [comment]
If you have external dependencies, errors messages should reflect those correctly.

My suspicion is that CVXOPT is tied to a very specific (range of?) build version of the MS C++ 14 build tools and isn't correctly identifying sub-builds, or it's just defaulting to an incorrect error message anytime there is a compile failure.

**Comment by stormont** [comment]
It may not be directly tied, but there's definitely some sort of dependency on it that's failing in ways that none of my other packages do.

I got it working, but it definitely seems like something is wrong in the configured package lookup; the package it tries to pull with `pip` doesn't work, but does if specified during the creation of a Conda environment.

Here's the different variations that I tried (and saw all fail with the same result):

0. (All using a fresh Anaconda environment and a `requirements.txt` that specifies `cvxopt>=1.1.9`)
1. (FAILED) Install with Python 3.6.5
3. (SUCCEEDED) Created a new Conda environment with `conda create -n temp python cvxopt`; this installed `1.1.9-py35_0`; this generated a Python 3.5.5 environment
4. (FAILED) Install with Python 3.5.5 inside the environment from (3), using a `requirements.txt` that specifies `cvxopt>=1.1.9`
5. (SUCCEEDED) Removed the 1.1.9 dependency from the `requirements.txt` to simply list `cvxopt`, causing the Conda-installed package to be used.

Here's the failing install from a fresh Python 3.6.5 Conda environment:

```
(D:\Programs\Anaconda3) D:\repos\third_party\temp>conda create -n temp python
Fetching package metadata ....

It is possible you have given conda an invalid channel. Please double-check
your conda configuration using `conda config --show`.

If the requested url is in fact a valid conda channel, please request that the
channel administrator create `noarch/repodata.json` and associated
`noarch/repodata.json.bz2` files, even if `noarch/repodata.json` is empty.
$ mkdir noarch
$ echo '{}' > noarch/repodata.json
$ bzip2 -k noarch/repodata.json
...........
Solving package specifications: .

Package plan for installation in environment D:\Programs\Anaconda3\envs\temp:

The following NEW packages will be INSTALLED:

    certifi:        2018.1.18-py36_0
    pip:            9.0.3-py36_0
    python:         3.6.5-h0c2934d_0
    setuptools:     39.0.1-py36_0
    vc:             14-h0510ff6_3
    vs2015_runtime: 14.0.25123-3
    wheel:          0.31.0-py36_0
    wincertstore:   0.2-py36h7fe50ca_0

Proceed ([y]/n)? y

#
# To activate this environment, use:
# > activate temp
#
# To deactivate an active environment, use:
# > deactivate
#
# * for power-users using bash, you must source
#


(D:\Programs\Anaconda3) D:\repos\third_party\temp>activate temp

(temp) D:\repos\third_party\temp>pip install -r requirements.txt
Collecting tensorflow>=1.0.0 (from -r requirements.txt (line 1))
  Using cached tensorflow-1.7.0-cp36-cp36m-win_amd64.whl
Collecting tflearn>=0.3.2 (from -r requirements.txt (line 2))
Collecting pympler>=0.5 (from -r requirements.txt (line 3))
Collecting cvxopt>=1.1.9 (from -r requirements.txt (line 4))
  Using cached cvxopt-1.1.9.tar.gz
Collecting seaborn>=0.8.1 (from -r requirements.txt (line 5))
Collecting pandas>=0.20.3 (from -r requirements.txt (line 6))
  Using cached pandas-0.22.0-cp36-cp36m-win_amd64.whl
Collecting six>=1.10.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached six-1.11.0-py2.py3-none-any.whl
Requirement already satisfied: wheel>=0.26 in d:\programs\anaconda3\envs\temp\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting astor>=0.6.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached astor-0.6.2-py2.py3-none-any.whl
Collecting absl-py>=0.1.6 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting protobuf>=3.4.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached protobuf-3.5.2.post1-cp36-cp36m-win_amd64.whl
Collecting numpy>=1.13.3 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached numpy-1.14.2-cp36-none-win_amd64.whl
Collecting grpcio>=1.8.6 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached grpcio-1.10.1-cp36-cp36m-win_amd64.whl
Collecting tensorboard<1.8.0,>=1.7.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached tensorboard-1.7.0-py3-none-any.whl
Collecting gast>=0.2.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting termcolor>=1.1.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting Pillow (from tflearn>=0.3.2->-r requirements.txt (line 2))
  Using cached Pillow-5.1.0-cp36-cp36m-win_amd64.whl
Collecting matplotlib (from seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached matplotlib-2.2.2-cp36-cp36m-win_amd64.whl
Collecting scipy (from seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached scipy-1.0.1-cp36-none-win_amd64.whl
Collecting pytz>=2011k (from pandas>=0.20.3->-r requirements.txt (line 6))
  Using cached pytz-2018.3-py2.py3-none-any.whl
Collecting python-dateutil>=2 (from pandas>=0.20.3->-r requirements.txt (line 6))
  Using cached python_dateutil-2.7.2-py2.py3-none-any.whl
Requirement already satisfied: setuptools in d:\programs\anaconda3\envs\temp\lib\site-packages (from protobuf>=3.4.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting bleach==1.5.0 (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached bleach-1.5.0-py2.py3-none-any.whl
Collecting html5lib==0.9999999 (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting werkzeug>=0.11.10 (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached Werkzeug-0.14.1-py2.py3-none-any.whl
Collecting markdown>=2.6.8 (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached Markdown-2.6.11-py2.py3-none-any.whl
Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached pyparsing-2.2.0-py2.py3-none-any.whl
Collecting cycler>=0.10 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached cycler-0.10.0-py2.py3-none-any.whl
Collecting kiwisolver>=1.0.1 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached kiwisolver-1.0.1-cp36-none-win_amd64.whl
Building wheels for collected packages: cvxopt
  Running setup.py bdist_wheel for cvxopt ... error
  Complete output from command D:\Programs\Anaconda3\envs\temp\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\devan\\AppData\\Local\\Temp\\pip-build-i9f7e89d\\cvxopt\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d C:\Users\devan\AppData\Local\Temp\tmpilsazim0pip-wheel- --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.6
  creating build\lib.win-amd64-3.6\cvxopt
  copying src\python\coneprog.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\cvxprog.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\info.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\misc.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\modeling.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\msk.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\printing.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\solvers.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\_version.py -> build\lib.win-amd64-3.6\cvxopt
  copying src\python\__init__.py -> build\lib.win-amd64-3.6\cvxopt
  UPDATING build\lib.win-amd64-3.6\cvxopt/_version.py
  set build\lib.win-amd64-3.6\cvxopt/_version.py to '1.1.9'
  running build_ext
  building 'base' extension
  error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

  ----------------------------------------
  Failed building wheel for cvxopt
  Running setup.py clean for cvxopt
Failed to build cvxopt
Installing collected packages: six, astor, absl-py, protobuf, numpy, grpcio, html5lib, bleach, werkzeug, markdown, tensorboard, gast, termcolor, tensorflow, Pillow, tflearn, pympler, cvxopt, pyparsing, python-dateutil, cycler, pytz, kiwisolver, matplotlib, scipy, pandas, seaborn
  Running setup.py install for cvxopt ... error
    Complete output from command D:\Programs\Anaconda3\envs\temp\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\devan\\AppData\\Local\\Temp\\pip-build-i9f7e89d\\cvxopt\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\devan\AppData\Local\Temp\pip-xww83lwz-record\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.6
    creating build\lib.win-amd64-3.6\cvxopt
    copying src\python\coneprog.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\cvxprog.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\info.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\misc.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\modeling.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\msk.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\printing.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\solvers.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\_version.py -> build\lib.win-amd64-3.6\cvxopt
    copying src\python\__init__.py -> build\lib.win-amd64-3.6\cvxopt
    UPDATING build\lib.win-amd64-3.6\cvxopt/_version.py
    set build\lib.win-amd64-3.6\cvxopt/_version.py to '1.1.9'
    running build_ext
    building 'base' extension
    error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

    ----------------------------------------
Command "D:\Programs\Anaconda3\envs\temp\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\devan\\AppData\\Local\\Temp\\pip-build-i9f7e89d\\cvxopt\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\devan\AppData\Local\Temp\pip-xww83lwz-record\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\devan\AppData\Local\Temp\pip-build-i9f7e89d\cvxopt\
```

Here's the Python 3.5.5 environment that succeeded installing the Conda package, failed to install from `requirements.txt` with the tightened restriction, then succeeded upon loosening the version restriction:

```
(temp) D:\repos\third_party\temp2>conda create -n temp2 python cvxopt
Fetching package metadata ....

It is possible you have given conda an invalid channel. Please double-check
your conda configuration using `conda config --show`.

If the requested url is in fact a valid conda channel, please request that the
channel administrator create `noarch/repodata.json` and associated
`noarch/repodata.json.bz2` files, even if `noarch/repodata.json` is empty.
$ mkdir noarch
$ echo '{}' > noarch/repodata.json
$ bzip2 -k noarch/repodata.json
...........
Solving package specifications: .

Package plan for installation in environment D:\Programs\Anaconda3\envs\temp2:

The following NEW packages will be INSTALLED:

    certifi:                 2018.1.18-py35_0
    cvxopt:                  1.1.9-py35_0
    intel-openmp:            2018.0.0-8
    m2w64-dsdp:              5.8-1
    m2w64-fftw:              3.3.4-6
    m2w64-gcc-libgfortran:   5.3.0-6
    m2w64-gcc-libs:          5.3.0-7
    m2w64-gcc-libs-core:     5.3.0-7
    m2w64-glpk:              4.61-1
    m2w64-gmp:               6.1.0-2
    m2w64-gsl:               2.1-2
    m2w64-libwinpthread-git: 5.0.0.4634.697f757-2
    m2w64-openblas:          0.2.19-1
    mkl:                     2017.0.4-h6d528fc_0
    msys2-conda-epoch:       20160418-1
    pip:                     9.0.3-py35_0
    python:                  3.5.5-h0c2934d_2
    setuptools:              39.0.1-py35_0
    vc:                      14-h0510ff6_3
    vs2015_runtime:          14.0.25123-3
    wheel:                   0.31.0-py35_0
    wincertstore:            0.2-py35hfebbdb8_0

Proceed ([y]/n)? y

intel-openmp-2 100% |###############################| Time: 0:00:00   3.50 MB/s
python-3.5.5-h 100% |###############################| Time: 0:00:03   5.17 MB/s
certifi-2018.1 100% |###############################| Time: 0:00:00   5.62 MB/s
setuptools-39. 100% |###############################| Time: 0:00:00   3.46 MB/s
wheel-0.31.0-p 100% |###############################| Time: 0:00:00   1.42 MB/s
pip-9.0.3-py35 100% |###############################| Time: 0:00:00   5.84 MB/s
#
# To activate this environment, use:
# > activate temp2
#
# To deactivate an active environment, use:
# > deactivate
#
# * for power-users using bash, you must source
#


(temp) D:\repos\third_party\temp2>activate temp2

(temp2) D:\repos\third_party\temp2>pip install -r requirements.txt
Collecting tensorflow>=1.0.0 (from -r requirements.txt (line 1))
  Downloading tensorflow-1.7.0-cp35-cp35m-win_amd64.whl (33.1MB)
    100% |################################| 33.1MB 35kB/s
Collecting tflearn>=0.3.2 (from -r requirements.txt (line 2))
  Using cached tflearn-0.3.2.tar.gz
Collecting pympler>=0.5 (from -r requirements.txt (line 3))
  Using cached Pympler-0.5.tar.gz
Collecting cvxopt>=1.1.9 (from -r requirements.txt (line 4))
  Using cached cvxopt-1.1.9.tar.gz
Collecting seaborn>=0.8.1 (from -r requirements.txt (line 5))
  Using cached seaborn-0.8.1.tar.gz
Collecting pandas>=0.20.3 (from -r requirements.txt (line 6))
  Downloading pandas-0.22.0-cp35-cp35m-win_amd64.whl (9.0MB)
    100% |################################| 9.0MB 98kB/s
Collecting termcolor>=1.1.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Cache entry deserialization failed, entry ignored
  Downloading termcolor-1.1.0.tar.gz
Collecting protobuf>=3.4.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Downloading protobuf-3.5.2.post1-cp35-cp35m-win_amd64.whl (958kB)
    100% |################################| 962kB 927kB/s
Collecting tensorboard<1.8.0,>=1.7.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached tensorboard-1.7.0-py3-none-any.whl
Collecting numpy>=1.13.3 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Downloading numpy-1.14.2-cp35-none-win_amd64.whl (13.4MB)
    100% |################################| 13.4MB 76kB/s
Requirement already satisfied: wheel>=0.26 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting gast>=0.2.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Cache entry deserialization failed, entry ignored
  Downloading gast-0.2.0.tar.gz
Collecting six>=1.10.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached six-1.11.0-py2.py3-none-any.whl
Collecting grpcio>=1.8.6 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Downloading grpcio-1.10.1-cp35-cp35m-win_amd64.whl (1.3MB)
    100% |################################| 1.3MB 567kB/s
Collecting astor>=0.6.0 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached astor-0.6.2-py2.py3-none-any.whl
Collecting absl-py>=0.1.6 (from tensorflow>=1.0.0->-r requirements.txt (line 1))
  Cache entry deserialization failed, entry ignored
  Downloading absl-py-0.1.13.tar.gz (80kB)
    100% |################################| 81kB 3.6MB/s
Collecting Pillow (from tflearn>=0.3.2->-r requirements.txt (line 2))
  Downloading Pillow-5.1.0-cp35-cp35m-win_amd64.whl (1.6MB)
    100% |################################| 1.6MB 615kB/s
Collecting scipy (from seaborn>=0.8.1->-r requirements.txt (line 5))
  Downloading scipy-1.0.1-cp35-none-win_amd64.whl (30.8MB)
    100% |################################| 30.8MB 36kB/s
Collecting matplotlib (from seaborn>=0.8.1->-r requirements.txt (line 5))
  Downloading matplotlib-2.2.2-cp35-cp35m-win_amd64.whl (8.7MB)
    100% |################################| 8.7MB 140kB/s
Collecting python-dateutil>=2 (from pandas>=0.20.3->-r requirements.txt (line 6))
  Using cached python_dateutil-2.7.2-py2.py3-none-any.whl
Collecting pytz>=2011k (from pandas>=0.20.3->-r requirements.txt (line 6))
  Using cached pytz-2018.3-py2.py3-none-any.whl
Requirement already satisfied: setuptools in d:\programs\anaconda3\envs\temp2\lib\site-packages (from protobuf>=3.4.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting html5lib==0.9999999 (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting markdown>=2.6.8 (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached Markdown-2.6.11-py2.py3-none-any.whl
Collecting bleach==1.5.0 (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached bleach-1.5.0-py2.py3-none-any.whl
Collecting werkzeug>=0.11.10 (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
  Using cached Werkzeug-0.14.1-py2.py3-none-any.whl
Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached pyparsing-2.2.0-py2.py3-none-any.whl
Collecting kiwisolver>=1.0.1 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Downloading kiwisolver-1.0.1-cp35-none-win_amd64.whl (57kB)
    100% |################################| 61kB 3.4MB/s
Collecting cycler>=0.10 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached cycler-0.10.0-py2.py3-none-any.whl
Building wheels for collected packages: tflearn, pympler, cvxopt, seaborn, termcolor, gast, absl-py
  Running setup.py bdist_wheel for tflearn ... done
  Stored in directory: C:\Users\devan\AppData\Local\pip\Cache\wheels\fb\06\72\0478c938ca315c6fddcce8233b80ec91a115ce4496a27e8c90
  Running setup.py bdist_wheel for pympler ... done
  Stored in directory: C:\Users\devan\AppData\Local\pip\Cache\wheels\57\c6\8b\91d4e0ffb106e935ca145964d13c03619943430ba9344c0206
  Running setup.py bdist_wheel for cvxopt ... error
  Complete output from command D:\Programs\Anaconda3\envs\temp2\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\devan\\AppData\\Local\\Temp\\pip-build-mt2op_sm\\cvxopt\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d C:\Users\devan\AppData\Local\Temp\tmp30hoh2h_pip-wheel- --python-tag cp35:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.5
  creating build\lib.win-amd64-3.5\cvxopt
  copying src\python\coneprog.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\cvxprog.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\info.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\misc.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\modeling.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\msk.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\printing.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\solvers.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\_version.py -> build\lib.win-amd64-3.5\cvxopt
  copying src\python\__init__.py -> build\lib.win-amd64-3.5\cvxopt
  UPDATING build\lib.win-amd64-3.5\cvxopt/_version.py
  set build\lib.win-amd64-3.5\cvxopt/_version.py to '1.1.9'
  running build_ext
  building 'base' extension
  error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

  ----------------------------------------
  Failed building wheel for cvxopt
  Running setup.py clean for cvxopt
  Running setup.py bdist_wheel for seaborn ... done
  Stored in directory: C:\Users\devan\AppData\Local\pip\Cache\wheels\29\af\4b\ac6b04ec3e2da1a450e74c6a0e86ade83807b4aaf40466ecda
  Running setup.py bdist_wheel for termcolor ... done
  Stored in directory: C:\Users\devan\AppData\Local\pip\Cache\wheels\de\f7\bf\1bcac7bf30549e6a4957382e2ecab04c88e513117207067b03
  Running setup.py bdist_wheel for gast ... done
  Stored in directory: C:\Users\devan\AppData\Local\pip\Cache\wheels\8e\fa\d6\77dd17d18ea23fd7b860e02623d27c1be451521af40dd4a13e
  Running setup.py bdist_wheel for absl-py ... done
  Stored in directory: C:\Users\devan\AppData\Local\pip\Cache\wheels\76\f7\0c\88796d7212af59bb2f496b12267e0605f205170781e9b86479
Successfully built tflearn pympler seaborn termcolor gast absl-py
Failed to build cvxopt
Installing collected packages: termcolor, six, protobuf, html5lib, markdown, numpy, bleach, werkzeug, tensorboard, gast, grpcio, astor, absl-py, tensorflow, Pillow, tflearn, pympler, cvxopt, scipy, pyparsing, kiwisolver, python-dateutil, cycler, pytz, matplotlib, pandas, seaborn
  Found existing installation: cvxopt 0+unknown
    Uninstalling cvxopt-0+unknown:
      Successfully uninstalled cvxopt-0+unknown
  Running setup.py install for cvxopt ... error
    Complete output from command D:\Programs\Anaconda3\envs\temp2\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\devan\\AppData\\Local\\Temp\\pip-build-mt2op_sm\\cvxopt\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\devan\AppData\Local\Temp\pip-p3q2kz9c-record\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.5
    creating build\lib.win-amd64-3.5\cvxopt
    copying src\python\coneprog.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\cvxprog.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\info.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\misc.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\modeling.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\msk.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\printing.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\solvers.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\_version.py -> build\lib.win-amd64-3.5\cvxopt
    copying src\python\__init__.py -> build\lib.win-amd64-3.5\cvxopt
    UPDATING build\lib.win-amd64-3.5\cvxopt/_version.py
    set build\lib.win-amd64-3.5\cvxopt/_version.py to '1.1.9'
    running build_ext
    building 'base' extension
    error: Microsoft Visual C++ 14.0 is required. Get it with "Microsoft Visual C++ Build Tools": http://landinghub.visualstudio.com/visual-cpp-build-tools

    ----------------------------------------
  Rolling back uninstall of cvxopt
Command "D:\Programs\Anaconda3\envs\temp2\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\devan\\AppData\\Local\\Temp\\pip-build-mt2op_sm\\cvxopt\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\devan\AppData\Local\Temp\pip-p3q2kz9c-record\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\devan\AppData\Local\Temp\pip-build-mt2op_sm\cvxopt\

(temp2) D:\repos\third_party\temp2>python --version
Python 3.5.5 :: Anaconda, Inc.

[NOTE: I manually removed the requirements.txt version restriction at this point)

(temp2) D:\repos\third_party\temp2>pip install -r requirements.txt
Requirement already satisfied: tensorflow>=1.0.0 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from -r requirements.txt (line 1))
Requirement already satisfied: tflearn>=0.3.2 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from -r requirements.txt (line 2))
Requirement already satisfied: pympler>=0.5 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from -r requirements.txt (line 3))
Requirement already satisfied: cvxopt in d:\programs\anaconda3\envs\temp2\lib\site-packages (from -r requirements.txt (line 4))
Collecting seaborn>=0.8.1 (from -r requirements.txt (line 5))
Collecting pandas>=0.20.3 (from -r requirements.txt (line 6))
  Using cached pandas-0.22.0-cp35-cp35m-win_amd64.whl
Requirement already satisfied: astor>=0.6.0 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: absl-py>=0.1.6 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: termcolor>=1.1.0 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: wheel>=0.26 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: six>=1.10.0 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: gast>=0.2.0 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: grpcio>=1.8.6 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: numpy>=1.13.3 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: tensorboard<1.8.0,>=1.7.0 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: protobuf>=3.4.0 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: Pillow in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tflearn>=0.3.2->-r requirements.txt (line 2))
Collecting matplotlib (from seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached matplotlib-2.2.2-cp35-cp35m-win_amd64.whl
Collecting scipy (from seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached scipy-1.0.1-cp35-none-win_amd64.whl
Collecting pytz>=2011k (from pandas>=0.20.3->-r requirements.txt (line 6))
  Using cached pytz-2018.3-py2.py3-none-any.whl
Collecting python-dateutil>=2 (from pandas>=0.20.3->-r requirements.txt (line 6))
  Using cached python_dateutil-2.7.2-py2.py3-none-any.whl
Requirement already satisfied: werkzeug>=0.11.10 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: markdown>=2.6.8 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: html5lib==0.9999999 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: bleach==1.5.0 in d:\programs\anaconda3\envs\temp2\lib\site-packages (from tensorboard<1.8.0,>=1.7.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Requirement already satisfied: setuptools in d:\programs\anaconda3\envs\temp2\lib\site-packages (from protobuf>=3.4.0->tensorflow>=1.0.0->-r requirements.txt (line 1))
Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached pyparsing-2.2.0-py2.py3-none-any.whl
Collecting cycler>=0.10 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached cycler-0.10.0-py2.py3-none-any.whl
Collecting kiwisolver>=1.0.1 (from matplotlib->seaborn>=0.8.1->-r requirements.txt (line 5))
  Using cached kiwisolver-1.0.1-cp35-none-win_amd64.whl
Installing collected packages: pytz, python-dateutil, pandas, pyparsing, cycler, kiwisolver, matplotlib, scipy, seaborn
Successfully installed cycler-0.10.0 kiwisolver-1.0.1 matplotlib-2.2.2 pandas-0.22.0 pyparsing-2.2.0 python-dateutil-2.7.2 pytz-2018.3 scipy-1.0.1 seaborn-0.8.1
```

**Comment by martinandersen** [comment]
CVXOPT 1.2.0 is now available for Windows and Python 3.5+ as a conda package.
